src/components/movieDetails/movieDetails.component.tsx

```diff
@@ -1,17 +1,35 @@
-import React, { FC } from 'react';
+import ROUTES from '@app/routes';
+import { useStoreon } from 'storeon/react';
+import React, { FC, useEffect } from 'react';
 import Logo from '@app/components/logo/logo.component';
 import useCommonStyle from '@app/style/variables/sizes';
+import { useHistory, useParams } from 'react-router-dom';
 import { faSearch } from '@fortawesome/free-solid-svg-icons';
 import { NO_IMAGE_PATH } from '@shared/interfaces/movies.model';
+import { ActionType, IState, StoreModule } from '@app/store/store.interface';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import useGenreStyle from '@app/components/movieItem/movieItem.component.style';
-import IMovieDetails from '@app/components/movieDetails/movieDetails.interface';
 import useStyle from '@app/components/movieDetails/movieDetails.component.style';
 
-const MovieDetails: FC<IMovieDetails> = ({ movie, onBackButtonClick }) => {
+const MovieDetails: FC = () => {
   const s = useStyle();
+  const history = useHistory();
   const genreStyles = useGenreStyle();
   const { appContainer } = useCommonStyle();
+  const { id } = useParams<{ id: string }>();
+  const { dispatch, search: { chosenMovie: movie, text } } = useStoreon<IState>(StoreModule.search);
+
+  useEffect(() => {
+    const isEmptyOrNewMovie = !movie || movie.id !== Number(id);
+
+    if (text && isEmptyOrNewMovie) {
+      dispatch(ActionType.getMovieDetails, id);
+    }
+
+    if (!text) {
+      history.replace(ROUTES.HOME);
```

Comment:
could you please add a proper routing here? Right now if you reload a movie page it redirects to home page. This remove all the benefits of routing

Reply:
Indeed, and it was made intentionally since no requirements were specified in the task related to the /movie/:id route for page reloading behavior. I decided to start user experience from scratch if the users tries to open this page without doing a search before. It was the easiest thing to implement that time, but now I've changed it a bit. It may cause some minor bugs now, hope you'll forgive me for them :)